src/bundle/DependencyInjection/Configuration/Parser/SiteAccessAware/AutocompleteParser.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Bundle\Search\DependencyInjection\Configuration\Parser\SiteAccessAware;
+
+use Ibexa\Bundle\Core\DependencyInjection\Configuration\AbstractParser;
+use Ibexa\Bundle\Core\DependencyInjection\Configuration\SiteAccessAware\ContextualizerInterface;
+use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;
+use Symfony\Component\Config\Definition\Builder\NodeBuilder;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+
+/**
+ * Configuration parser for search.
+ *
+ * @Example configuration:
+ *
+ * ```yaml
+ * ibexa:
+ *   system:
+ *      default: # configuration per siteaccess or siteaccess group
+ *          search:
+ *              autocomplete:
+ *                  min_search_test_length: 3
+ *                  result_limit: 5
+ * ```
+ */
+final class AutocompleteParser extends AbstractParser
+{
+    /**
+     * @param array<string, mixed> $scopeSettings
+     */
+    public function mapConfig(
+        array &$scopeSettings,
+        $currentScope,
+        ContextualizerInterface $contextualizer
+    ): void {
+        if (empty($scopeSettings['search'])) {
+            return;
+        }
+
+        $settings = $scopeSettings['search'];
+
+        $this->addAutocompleteParameters($settings, $currentScope, $contextualizer);
+    }
+
+    public function addSemanticConfig(NodeBuilder $nodeBuilder): void
+    {
+        $rootProductCatalogNode = $nodeBuilder->arrayNode('search');
+        $rootProductCatalogNode->append($this->addAutocompleteConfiguration());
+    }
+
+    /**
+     * @param array<string, mixed> $settings
+     */
+    private function addAutocompleteParameters(
+        $settings,
+        string $currentScope,
+        ContextualizerInterface $contextualizer
+    ): void {
+        $names = [
+            'min_search_test_length',
+            'result_limit',
+        ];
+        foreach ($names as $name) {
+            if (isset($settings['autocomplete'][$name])) {
+                $contextualizer->setContextualParameter(
+                    'search.autocomplete.' . $name,
+                    $currentScope,
+                    $settings['autocomplete'][$name]
+                );
+            }
+        }
+    }
+
+    private function addAutocompleteConfiguration(): ArrayNodeDefinition
+    {
+        $treeBuilder = new TreeBuilder('autocomplete');
+        $node = $treeBuilder->getRootNode();
+
+        $node
+            ->children()
+                ->integerNode('min_search_test_length')
```

Comment:
Can we consider alternatives to `min_search_test_length` key ? For instance

* `min_query_length`
* `min_phrase_length`
* `min_search_phrase_length`

IMHO `min_search_test_length` is not the best naming choice